src/visualMarkers/main.ts

```diff
@@ -0,0 +1,117 @@
+import fuzzysort from 'fuzzysort'
+import {
+  Events,
+  EventType,
+  MarkChangeEvent,
+} from '../events'
+import { illumination, removeMapMark, signalFlare, smoke } from '../trigger'
+import { getMarkById } from '../custom'
+import { CommandType as EventCommandType } from '../commands/types'
+import { randomAngleDeg } from '../common'
+
+// const colorArray = Object.keys(SmokeColor).filter((v) => isNaN(Number(v))) can be used alternatively, but uses DCS smoke color names
+const smokeColor:string[] = [
+  'unspecified',
+  'green',
+  'red',
+  'white',
+  'orange',
+  'blue'
+] 
+
+const flareColor:string[] = [
+  'unspecified',
+  'green',
+  'red',
+  'white',
+  'yellow'
+] 
+
+export async function main(): Promise<() => Promise<void>> {
+    const subscription = Events.subscribe(async event => {
+      if (EventType.MarkChange === event.type) {
+        return handleMarkChangeEvent(event as MarkChangeEvent)
+      }
+    })
+
+  return async () => {
+    subscription.unsubscribe()
+  }
+}
+
+async function handleMarkChangeEvent(event: MarkChangeEvent) {
+  const { id, command } = event
+  // attempt to handle command(s) from markers
+  
+  if (command) {
+    if (EventCommandType.Smoke === command.type) {
+      const addedMark = await getMarkById(id)
+
+      if (!addedMark) {
+        throw new Error('expected addedMark')
+      }
+
+      let color = 1 //default green smoke
```

Comment:
lets use an enum here, having to guess what a number means is bad for velocity

Reply:
changed to enum in https://github.com/bitbubbler/yet-another-dcs-server/pull/13/commits/c739f19620b4bd3fa549c0263fe09e99040170f6